File: Hackhthon-II/phase-II/frontend/src/types/task.ts
/**
 * TypeScript types for Todo Application.
 */

/**
 * Task entity representing a single to-do item.
 */
export interface Task {
  id: number;
  user_id: string;
  title: string;
  description: string | null;
  completed: boolean;
  created_at: string;
  updated_at: string;
}

/**
 * Request payload for creating a new task.
 */
export interface TaskCreateRequest {
  title: string;
  description?: string;
}

/**
 * Request payload for updating an existing task.
 */
export interface TaskUpdateRequest {
  title?: string;
  description?: string;
}

/**
 * Paginated task list response.
 */
export interface TaskListResponse {
  tasks: Task[];
  total: number;
  offset: number;
  limit: number;
}

File: backend/src/models/task.py
"""
Task model for Todo Application.
"""
from datetime import datetime
from sqlmodel import SQLModel, Field, Relationship
from typing import Optional


class Task(SQLModel, table=True):
    """
    Represents a single to-do item owned by a user.

    Attributes:
        id: Unique task identifier (auto-increment)
        user_id: Foreign key to users table (owner)
        title: Task title (required, 1-200 characters)
        description: Optional task details (max 1000 characters)
        completed: Task completion status (default: False)
        created_at: Task creation timestamp
        updated_at: Last modification timestamp
    """
    __tablename__ = "tasks"

    id: Optional[int] = Field(default=None, primary_key=True)
    user_id: str = Field(foreign_key="users.id")
    title: str = Field(max_length=200)
    description: Optional[str] = Field(default=None, max_length=1000)
    completed: bool = Field(default=False, index=True)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    # Relationships
    user: "User" = Relationship(back_populates="tasks")
